dll/win32/shell32/CDefView.cpp

```diff
@@ -160,6 +161,7 @@ class CDefView :
 
     BOOL                      m_isEditing;
     BOOL                      m_isParentFolderSpecial;
+    BYTE                      m_ScheduledStatusbarUpdate;
```

Comment:
Would it make sense to use an ULONG/LONG and interlocked functions to modify this?

Reply:
Like most of the shell, DefView is supposed to be a STA object. Even if it was not, this variable is only modified on the ListView thread because it does its work in a ListView notification.

Reply:
Is this m_ScheduledStatusbarUpdate meant to have values strictly greater than 1 too?

Reply:
Not as it stands. This itemchanged notification is by far the worst offender of triggering many useless updates but I can imagine there being other places where you want to do some work before allowing an update at the end.

Changing this to bool and = true gains us nothing.

Reply:
Well then actually, please change it to BOOL and TRUE . Please avoid dubious optimizations that don't optimize anything and make code less clear (see why I asked the question. If it was BOOL, FALSE/TRUE it would have been clear from the very beginning.)

Reply:
There are several methods in DefView that are UNIMPLEMENTED; that might cause unwanted updates. On this variable you would do ++ when you start and -- at the end of processing.

But I'll change it to true/false now and deal with the ++ stuff when it becomes needed later.

Reply:
Ah ok thanks it makes sense (with the ++ and --) . Let's then proceed with whatever you think is more suited.